lib/horde/supervisor.ex

```diff
@@ -25,8 +25,16 @@ defmodule Horde.Supervisor do
               shutting_down: false
   end
 
-  def child_spec(id) do
-    %{id: id, start: {__MODULE__, :start_link, [id]}}
+  def child_spec(options \\ []) do
+    options =
+      Keyword.put_new(options, :id, __MODULE__)
+      |> Keyword.put_new(:children, [])
```

Comment:
Not important, but I find this style a bit more readable
```
options =
  options
  |> Keyword.put_new(:id, __MODULE__)
  |> Keyword.put_new(:children, [])
```

Reply:
perhaps use Keyword.merge/2:
`options = Keyword.merge([id: __MODULE__, children: []], options)`